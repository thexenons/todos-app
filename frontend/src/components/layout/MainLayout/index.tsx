import { FC, PropsWithChildren, Suspense } from "react";
import Footer from "../Footer";
import Header from "../Header";

const MainLayout: FC<PropsWithChildren<{}>> = ({ children }) => (
	<>
		<Header />
		<main>
			<Suspense fallback={<div>Loading...</div>}>{children}</Suspense>
		</main>
		<Footer />
	</>
);

export default MainLayout;
