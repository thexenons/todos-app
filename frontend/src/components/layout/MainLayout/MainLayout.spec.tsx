import { render } from "@testing-library/react";

import BaseTestComponent from "../../../tests/BaseTestComponent";
import MainLayout from ".";

describe("MainLayout", () => {
	it("Render", () => {
		const { container } = render(
			<BaseTestComponent>
				<MainLayout />
			</BaseTestComponent>
		);
		expect(container).toMatchSnapshot();
	});
});
