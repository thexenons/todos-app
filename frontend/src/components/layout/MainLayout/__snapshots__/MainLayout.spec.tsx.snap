// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainLayout > Render 1`] = `
<div>
  <aside
    class="styled__SidebarWrapper-sc-d583294-4 knaZwA"
  >
    <button
      class="styled__ButtonWrapper-sc-f8cd241f-0 dlIeAl styled__SidebarButton-sc-d583294-0 egJNPm"
    >
      <svg
        class="styled__SidebarButtonContent-sc-d583294-1 iybvkH"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
    </button>
    <div
      class="styled__SidebarContentWrapper-sc-d583294-2 drDxhd"
    >
      <div
        class="styled__SidebarContent-sc-d583294-3 enpdwx"
      >
        <div>
          <div>
            <a
              href="/login"
            >
              Login
            </a>
          </div>
        </div>
      </div>
    </div>
  </aside>
  <main />
</div>
`;
